Add external instructions for running inferCNV (#1034)
AlexsLemonade/scpca-nf: external-instructions.md

## CNV inference

`scpca-nf` currently performs copy-number variation (CNV) inference using [`inferCNV`](https://github.com/broadinstitute/infercnv).

By default, CNV inference is not performed.
As part of its algorithm, `inferCNV` uses a set of normal reference cells to quantify CNV events across all provided cells.
`scpca-nf` uses results from its cell type annotation module to create this normal reference.
As such, to perform CNV inference, you must also [perform cell type annotation, as described above](#cell-type-annotation).
Note that if CNV inference is turned on, cell type annotation will automatically be turned on as well.

You can turn on CNV inference by taking the following steps:

1. Ensure your [prepared sample metadata file](#prepare-the-sample-metadata-file) contains a `diagnosis` column.
   This column is **required** in your sample metadata file as `scpca-nf` uses this information to create the normal reference.
2. [Determine which cell types](#determining-normal-reference-cell-types) should be used for the `inferCNV` normal reference.

> 3. [Prepare the diagnosis metadata files](#preparing-diagnosis-metadata-files) which the workflow will use to create the normal cell type reference.Using the information from the previous steps,

Reviewer: ```suggestion
3. [Prepare the diagnosis metadata files](#preparing-diagnosis-metadata-files) which the workflow will use to create the normal cell type reference using the information from the previous steps. 
```

>    You will need to provide the path/uri to this files as a workflow parameters, which you will need to define in your configuration file.

Reviewer: ```suggestion
   You will need to provide the path/uri to these files as workflow parameters, which you will need to define in your configuration file.
```

   For more information on adding parameters to your configuration file, see [Configuring scpca-nf for your environment](#configuring-scpca-nf-for-your-environment).
4. Run the workflow with the `--perform_cnv_inference` flag.

> Once you have followed both the above steps and the [steps to prepare for cell type annotation](#cell-type-annotation) and you have added the path/uri to the diagnosis metadata files to your configuration file, you can use the following command to run the workflow with CNV inference:

Reviewer: ```suggestion
Once you have followed the above steps, including the [steps to prepare for cell type annotation](#cell-type-annotation), and you have added the path/uri to the diagnosis metadata files to your configuration file, you can use the following command to run the workflow with CNV inference:
```

```sh
nextflow run AlexsLemonade/scpca-nf \
  --perform_cnv_inference
```

### Prepare the diagnosis metadata files

`scpca-nf` uses the consensus cell type labels derived from `SingleR` and `CellAssign` annotations to create the set of normal reference cells.
The specific consensus cell types to use are determined by the sample's diagnosis.

To manage many diagnoses and consensus cell types, the Data Lab has organized [individual sample diagnoses into broad diagnosis groups](references/broad-diagnosis-map.tsv), and similarly [consensus cell types into validation groups](references/consensus-validation-groups.tsv).

> `scpca-nf` uses these grouping to specify the normal reference cell types as follows:

Reviewer: ```suggestion
`scpca-nf` uses these groupings to specify the normal reference cell types as follows:
```

1. Given a sample's diagnosis, identify the corresponding broad diagnosis

>     * This step utilizes the provided `params.diagnosis_groups_file` TSV

Reviewer: A thought I had while reading this is that we could make this file optional and if that's the case, then they just need to make sure the diagnosis in the sample metadata table matches the diagnosis in the column for broad diagnosis in `diagnosis_celltypes_file`. I feel like the broad groups for diagnoses are really just helpful for us and not for external users that are running samples independently. 

If we did that, then we should simplify all of these instructions to not even mention the broad diagnosis file.

Reviewer: Related to this, maybe we should just refocus this on how to run inferCNV using our existing metadata files and then as a future TODO, ensure we are able to support someone running the workflow with a diagnoses that we don't have or a custom normal reference list.

Author: > A thought I had while reading this is that we could make this file optional and if that's the case, then they just need to make sure the diagnosis in the sample metadata table matches the diagnosis in the column for broad diagnosis in diagnosis_celltypes_file. I feel like the broad groups for diagnoses are really just helpful for us and not for external users that are running samples independently.

In a scenario where this file is optional, it sounds like you still envision _us_ using the file?

2. Identify the consensus cell type validation groups that are expected to be normal cells in samples of that diagnosis
    * This step utilizes the provided `params.diagnosis_celltypes_file` TSV
3. Identify the individual consensus cell types that comprise those validation groups

To support the first two steps of this procedure, you will need to provide two metadata TSV files:

* [The diagnosis groups metadata file](#diagnosis-groups-metadata-file), which maps individual sample diagnoses to broad diagnosis groups
* [The diagnosis cell types metadata file](#diagnosis-cell-types-metadata-file), maps broad diagnosis groups to consensus cell type validation groups

The Data Lab has prepared these metadata files for all sample diagnoses present in the ScPCA Portal, which `scpca-nf` uses by default:

* The diagnosis groups metadata file is provided in [`references/broad-diagnosis-map.tsv`](references/broad-diagnosis-map.tsv)
  * Broad diagnoses are listed in the `diagnosis_group` column, and sample-level diagnoses are listed in the `submitted_diagnosis` column

> * The diagnosis cell types metadata file is provided in [`references/diagnosis-celltype-groups.tsv`](references/diagnosis-celltype-groups.tsv)

Reviewer: If we keep the instructions for using a custom diagnoses, we need to provide a list of all possible validation groups. I think we have a table that matches consensus to validation so we should direct them to that table.

  * Broad diagnoses are listed in the `diagnosis_group` column, and consensus cell type validation groups are listed in the listed in the `celltype_groups` column

You can inspect these files to determine if your sample's diagnosis is already present in the `submitted_diagnosis` column of the diagnosis groups metadata file.

> If your sample's diagnosis is present, you do not have to prepare your own metadata files; you can let `scpca-nf` make use of these existing metadata files.

Reviewer: I would move this statement to the very top of this section. I also might refocus this and have the first section be just how to run inferCNV using existing metadata files and then have these sections that talk about actually making the metadata files. They only need to make this file if they have a diagnoses that is not present yet in our data.
